opentelemetry-instrumentation/src/opentelemetry/instrumentation/auto_instrumentation/_load.py

```diff
@@ -58,23 +80,25 @@ def _load_distro() -> BaseDistro:
 
 def _load_instrumentors(distro):
     package_to_exclude = environ.get(OTEL_PYTHON_DISABLED_INSTRUMENTATIONS, [])
+    entry_point_finder = _EntryPointDistFinder()
     if isinstance(package_to_exclude, str):
         package_to_exclude = package_to_exclude.split(",")
         # to handle users entering "requests , flask" or "requests, flask" with spaces
         package_to_exclude = [x.strip() for x in package_to_exclude]
 
-    for entry_point in iter_entry_points("opentelemetry_pre_instrument"):
+    for entry_point in entry_points(group="opentelemetry_pre_instrument"):
         entry_point.load()()
 
-    for entry_point in iter_entry_points("opentelemetry_instrumentor"):
+    for entry_point in entry_points(group="opentelemetry_instrumentor"):
         if entry_point.name in package_to_exclude:
             _logger.debug(
                 "Instrumentation skipped for library %s", entry_point.name
             )
             continue
 
         try:
-            conflict = get_dist_dependency_conflicts(entry_point.dist)
+            entry_point_dist = entry_point_finder.dist_for(entry_point)
+            conflict = get_dist_dependency_conflicts(entry_point_dist)
```

Comment:
entry_point.dist used in [line 77](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/2181/files#diff-6ecd6cbdf97812bf55619474db901385d54a6a6931b91b6b2fc5b308b1fb8ab3R77) doesn't seem to exist in [older python versions](https://github.com/python/cpython/blob/3.8/Lib/importlib/metadata.py#L41) and is an optional attribute in the [latest version](https://github.com/python/cpython/blob/3.12/Lib/importlib/metadata/__init__.py#L194).

Reply:
If we are going to use importlib_metadata (via opentelemetry api util) we just need to be assured that it's there since 6+ no? Looking at some old pkg_resources code, dist was default None there too.

Reply:
We're not going to be guaranteed 6+ if https://github.com/open-telemetry/opentelemetry-python/pull/4177 goes through - we'll only have 6 on python 3.12 and higher.